packages/paste-core/components/combobox/src/Combobox.tsx

```diff
@@ -29,99 +29,104 @@ const StyledInputAsSelect = styled(InputElement)<ComboboxProps>(props =>
 );
 /* eslint-enable */
 
-const renderItem = ({
-  item,
-  index,
-  getItemProps,
-  highlightedIndex,
-  optionTemplate,
-  inGroup,
-  optionUID,
-}: RenderItemProps): React.ReactNode => {
+const Item: React.FC<ItemProps> = ({item, index, getItemProps, highlightedIndex, optionTemplate, inGroup}) => {
+  const UIDSeed = useUIDSeed();
   return (
     <ComboboxListboxOption
       {...getItemProps({item, index})}
       highlighted={highlightedIndex === index}
-      key={optionUID(index)}
+      key={UIDSeed(item)}
       variant={inGroup ? 'groupOption' : 'default'}
     >
       {optionTemplate ? optionTemplate(item) : item}
     </ComboboxListboxOption>
   );
 };
 
-const renderItems = ({
-  items,
-  getItemProps,
-  highlightedIndex,
-  optionTemplate,
-  optionUID,
-}: RenderItemsProps): React.ReactNode[] =>
-  items.map((item, index) => {
-    return renderItem({item, index, getItemProps, highlightedIndex, optionTemplate, optionUID});
-  });
+const Items: React.FC<ItemsProps> = ({items, getItemProps, highlightedIndex, optionTemplate}) => {
+  return (
+    <>
+      {items.map((item, index) => (
+        <Item
+          // eslint-disable-next-line react/no-array-index-key
+          key={index}
+          item={item}
+          index={index}
+          getItemProps={getItemProps}
+          highlightedIndex={highlightedIndex}
+          optionTemplate={optionTemplate}
+        />
+      ))}
+    </>
+  );
+};
 
-const renderGroupedItems = ({
+const GroupedItems: React.FC<GroupItemsProps> = ({
   items,
   getItemProps,
   highlightedIndex,
   optionTemplate,
   groupLabelTemplate,
   groupItemsBy,
-  optionUID,
-  groupUID,
-}: RenderGroupItemsProps): React.ReactNode[] | null => {
-  if (groupItemsBy != null) {
-    const groupedItems = groupBy(items, (item: {}) => item[groupItemsBy]);
-    let itemIndex = 0;
-    return Object.keys(groupedItems).map(group => {
-      if (group === 'undefined') {
-        return groupedItems[group].map((item: {}) => {
-          return renderItem({item, index: itemIndex++, getItemProps, highlightedIndex, optionTemplate, optionUID});
-        });
-      }
-      return (
-        <ComboboxListboxGroup groupName={group} groupLabelTemplate={groupLabelTemplate} key={groupUID(group)}>
-          {groupedItems[group].map((item: {}) => {
-            return renderItem({
-              item,
-              index: itemIndex++,
-              getItemProps,
-              highlightedIndex,
-              optionTemplate,
-              inGroup: true,
-              optionUID,
-            });
-          })}
-        </ComboboxListboxGroup>
-      );
-    });
+}) => {
+  const UIDSeed = useUIDSeed();
+
+  // Should never happen
+  if (groupItemsBy == null) {
+    return null;
   }
-  return null;
+
+  const groupedItems = groupBy(items, (item: {}) => item[groupItemsBy]);
+  const groupedItemKeys = Object.keys(groupedItems);
+
+  return (
+    <>
```

Comment:
Nit: don't necessarily need the fragment here either

Reply:
Because this component is optionally returning an array, we need to wrap it in a fragment. 

TS warning too:
![image](https://user-images.githubusercontent.com/1592327/96752226-2e724a00-139c-11eb-81cf-350d9d68788a.png)